src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -327,6 +328,32 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
             return Text(runner.GetFormattingInfo(), irContext, args, runner.CancellationToken);
         }
 
+        public static FormulaValue TextToBlob(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            runner.CancellationToken.ThrowIfCancellationRequested();
+
+            StringValue sv = (StringValue)args[0];
+            IResourceManager resourceManager = runner.FunctionServices.GetService<IResourceManager>();
+
+            if (resourceManager == null)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = $"Cannot convert {args[0].Type._type} to Blob, missing resource manager.",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.InvalidArgument
+                });
+            }
+
+            return new BlobValue(resourceManager, new StringResourceElement(resourceManager, sv.Value).Handle);
+        }
+
+        public static FormulaValue BlobToText(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            runner.CancellationToken.ThrowIfCancellationRequested();
+            return FormulaValue.New((args[0] as FileValue).ToString());

```

Comment:
>(args[0] as FileValue).ToString() [](http://example.com/codeflow?start=36&length=33)

Can we return an _identifier_ to the blob, so that if we need to convert it back to a blob, we can check if the resource manager has that value, and if so, return a reference to the same resource?

Reply:
This is what we're doing here
FileValue.ToString() => ResourceElement.Uri.ToString()